src/core/ddsc/src/dds_write.c

```diff
@@ -181,6 +185,37 @@ dds_return_t dds_write_impl (dds_writer *wr, const void * data, dds_time_t tstam
                    ((action & DDS_WR_UNREGISTER_BIT) ? NN_STATUSINFO_UNREGISTER : 0));
   d->timestamp.v = tstamp;
   ddsi_serdata_ref (d);
+
+#ifdef DDSI_INCLUDE_SHM
+  if (wr->m_entity.m_domain->gv.config.enable_shm && wr->m_wr->num_ice_proxy_reader)
+  {
+    struct iceoryx_header ice_hdr;
+    ddsrt_iovec_t iov;
+    uint32_t size = ddsi_serdata_size (d);
+    (void) ddsi_serdata_to_ser_ref (d, 0, size, &iov);
+    uint32_t send_size = (uint32_t) iov.iov_len;
+    char *send_ptr = iov.iov_base;
+    char *data_ptr;
+    while (1)
```

Comment:
In order not to send outdated data, I suggest limiting the time of attempts to send messages. Return the error code when the timeout expires.